### ISSUE TITLE
Call to a member function getUsername() on null

### ISSUE THREAD

**Issue by Skildar** [lexik/LexikJWTAuthenticationBundle#591]
Hi guyz,
I've been fiddling with this bundle for a few days, and I'm having a bit of a difficulty understanding how everything communicates with each other.
I'm using Symfony 3.4 and the latest release of the Lexik bundle but I'm stuck on a simple call to JWT getUser() inside my controller:
``` php
// src\AppBundle\Controller\UserController.php
<?php

namespace AppBundle\Controller;

use AppBundle\Entity\Account;

use Symfony\Bundle\FrameworkBundle\Controller\Controller;
use Symfony\Component\Routing\Annotation\Route;

use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpFoundation\Request;

use Symfony\Component\Security\Core\User\UserInterface;
use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;

class UserController extends Controller
{
...
  /**
   * Retrieve information about user
   * @Route("api/secured/user", methods={"GET"})
   */
  public function getUserInfo()
  {
    return new Response(sprintf('Logged in as %s', $this->getUser()->getUsername()));
  }
...
}
```

I based some of my configuration and code on https://github.com/chalasr/lexik-jwt-authentication-sandbox.

To make this easier I'll copy my security.yml :
``` yml
// app\config\security.yml
# To get started with security, check out the documentation:
# https://symfony.com/doc/current/security.html
security:

    encoders:
        Symfony\Component\Security\Core\User\User:
            algorithm: bcrypt
        AppBundle\Entity\Account:
            algorithm: bcrypt

    # https://symfony.com/doc/current/security.html#b-configuring-how-users-are-loaded
    providers:
        in_memory:
#            memory: ~
            memory:
                users:
                    ryan:
                        password: $2y$13$SbvhAV2T/SpwRl93f./zt.xD1CQCKYCgAzU7PBHjYMaI.kHSJ3Xnq
                        roles: 'ROLE_USER'
                    admin:
                        password: $2y$13$uGXOChJqXzvHr3lVuPzZqe9UCmdNy1wIBgsZfMZGIhVAGmV.AjDhy
                        roles: 'ROLE_ADMIN'
        entity_provider:
            entity:
                class: AppBundle:Account
                property: mail

    firewalls:
        # Lexik JWT Authentication
        login:
            pattern: ^/api/user/login
            stateless: true
            anonymous: true
            provider: entity_provider
            json_login:
                check_path:                 /api/user/login_check
                success_handler:            lexik_jwt_authentication.handler.authentication_success
                failure_handler:            lexik_jwt_authentication.handler.authentication_failure
                username_path:              email
                password_path:              password
                require_previous_session:   false

        api:
            # ...
            pattern: ^/api/secured
            stateless: true
            anonymous: false
            provider: entity_provider
            guard:
                authenticators:
                    - lexik_jwt_authentication.jwt_token_authenticator

        main:
            anonymous: ~
            provider: entity_provider

    access_control:
        - { path: ^/api/user, role: IS_AUTHENTICATED_ANONYMOUSLY }
        - { path: ^/api/secured/, roles: IS_AUTHENTICATED_FULLY }
```

After authentication succeeded, I add the token to the header of my request as described in the documentation. But when calling the route to retrieve more information about my current user, I get this error. The problem is, despite the authentication through my header JWT being a success, I feel like I missed a connection in my config so that my controller has access to the getUser() method. The documentation has failed to enlighten me on this point, could you shine some light on what I'm missing here?

**Comment by Skildar** [comment]
I cleared my cache and it gave me another error :
`Object of class AppBundle\Entity\Account could not be converted to string`

But a quick search on stackoverflow gave me directions on what to add to my Entity file.

``` php
// src\AppBundle\EntityAccount.php
    public function __toString() {
        return $this;
    }
```
I've not seen those lines required in the project I referenced to in my previous post, and yet it did exactly what was missing. I don't really get why that function didn't not exist and required me to explicit it.
